scripts/build/termux_step_get_dependencies.sh

```diff
@@ -65,6 +72,12 @@ termux_step_get_dependencies() {
 				continue
 			elif [ "$PKG" = "ERROR" ]; then
 				termux_error_exit "Obtaining buildorder failed"
+			elif [ "$PKG" = "ghc" ] || [ "$PKG" = "ghc-libs" ] || [ "$PKG" = "ghc-libs-static" ]; then
```

Comment:
There are currently very few of these package specific quirks in all of our build scripts, and it is probably a good idea to keep it that way. 

Running some sort of `termux_setup_ghc` in `termux_step_pre_configure` and getting the haskell dependencies there is more in line with how it is done for other packages. Instead of having IS_GHC_REQUIRED you would then run the setup function for those packages. 
A variable `TERMUX_PKG_GHC_REQUIRED` could also work.